db/change_cache.go

```diff
@@ -208,13 +212,36 @@ func (c *changeCache) Start(initialSequence uint64) error {
 	return nil
 }
 
+// waitForBGTCompletion waits for all the background tasks to finish.
+func waitForBGTCompletion(waitTime time.Duration, terminated []TaskStat) {
```

Comment:
This only gets called from locations where a databaseContext is available.  It would be preferable to make this a method on DatabaseContext, and then use the more standard InfofCtx logging.  This would also simplify TaskStat.

Reply:
Yeah, that is a good idea. But we won't still be able to make use of the standard InfofCtx logging since the context reference is not available in DatabaseContext. This change has been included in the latest revision.